vapi/simulator/simulator.go

```diff
@@ -2105,3 +2109,48 @@ func (s *handler) vmID(w http.ResponseWriter, r *http.Request) {
 func (s *handler) debugEcho(w http.ResponseWriter, r *http.Request) {
 	r.Write(w)
 }
+
+func (s *handler) shutdown(w http.ResponseWriter, r *http.Request) {
+	delayShutdown := func(delay int) {
+		go time.AfterFunc(time.Duration(delay)*time.Minute, func() {
+			s.Lock()
+			s.Config.ShutdownTime = ""
+			s.Config.Action = ""
+			s.Config.Reason = ""
+			s.Unlock()
+		})
+	}
+
+	switch r.Method {
+	case http.MethodGet:
+		OK(w, s.Config)
+	case http.MethodPost:
+		switch r.URL.Query().Get(shutdown.Action) {
+		case shutdown.Cancel:
+			delayShutdown(0)
+			w.WriteHeader(http.StatusNotFound)
```

Comment:
These should be `http.StatusNoContent` instead of NotFound, right?

Reply:
Sorry, Now i have updated to `w.WriteHeader(http.StatusNoContent)`